Core/source/core/pgp-key.ts

```diff
@@ -383,6 +380,16 @@ export class PgpKey {
       lastModified,
       revoked: k.revocationSignatures.length > 0
     };
+
+    if (k.isPrivate()) {
+      Object.assign(keyDetails, {
+        private: k.armor(),
+        isFullyDecrypted: isFullyDecrypted(k),
+        isFullyEncrypted: isFullyEncrypted(k)
+      });
```

Comment:
Maybe better to return spread object(I don't prefer object assign though)
https://eslint.org/docs/latest/rules/prefer-object-spread
```ts
return {
    ...keyDetails,
    {
        private: k.armor(),
        isFullyDecrypted: isFullyDecrypted(k),
        isFullyEncrypted: isFullyEncrypted(k)
  }
};

```

Reply:
I like spread operator too for it's simplicity, but here we'll need to have 2 `return` to make it work:
```ts
if (k.isPrivate()) {
    return {
        ...keyDetails, 
        {
             private: k.armor(), 
             isFullyDecrypted: isFullyDecrypted(k),
             isFullyEncrypted: isFullyEncrypted(k)
        } 
    }
} else {
    return keyDetails
}

```

Reply:
Yeah, we indeed need it (we don't need else though).
But it's not bad for code performance and readability i think?

Reply:
I'm fine with your approach though